src/charm.py

```diff
@@ -54,6 +56,21 @@ def __init__(self, *args):
         # --- LIB EVENT HANDLERS ---
         self.restart = RollingOpsManager(self, relation=RESTART_RELATION, callback=self._restart)
 
+        # cos agent
```

Comment:
I've been thinking about this one, I think it's fine keeping it here for the time being, but we may in the future (should we have a k8s charm) move this over an Observability Manager